community/modules/scheduler/schedmd-slurm-gcp-v6-controller/cleanup.tf

```diff
@@ -0,0 +1,35 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+resource "null_resource" "cleanup_compute" {
+  count = var.enable_cleanup_compute ? 1 : 0
+
+  triggers = {
+    project_id   = var.project_id
+    cluster_name = local.slurm_cluster_name
+  }
+
+  provisioner "local-exec" {
```

Comment:
Reading [this](https://developer.hashicorp.com/terraform/language/resources/provisioners/syntax#destroy-time-provisioners), it is possible to eliminate line 17 in favor of a dynamic provisioner block.

Something like

```hcl
  dynamic "provisioner" {
    count = var.enable_cleanup_compute ? 1 : 0
    labels = [ "local-exec" ]
    content {
      whatever
    }
```
  }
```

Reply:
https://discuss.hashicorp.com/t/how-to-dynamically-define-multiple-provisioner-remote-exec/17769

> There is no mechanism for dynamically configuring provisioners in Terraform. [Provisioners are a feature of last resort 92](https://www.terraform.io/docs/provisioners/index.html#provisioners-are-a-last-resort), so their functionality is limited.


On a bright side, after this PR majority of users will not need to touch `enable_cleanup_compute` -> reducing unpleasant encounters.